sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -508,55 +258,67 @@ public String getLabel() {
     @JsonGetter("linkUrl")
     @JsonInclude(JsonInclude.Include.NON_NULL)
     @Nullable
-    public String getLinkUrl() {
-      return url;
-    }
+    public abstract String getLinkUrl();
 
-    @Override
-    public String toString() {
-      return String.format("%s:%s=%s", ns, key, value);
+    public static <T> Item<T> create(String key, Type type, @Nullable T value) {
+      FormattedItemValue formatted = safeFormat(type, value);
+      return of(null, key, type, formatted.getLongValue(), formatted.getShortValue(), null, null);
     }
 
-    @Override
-    public boolean equals(Object obj) {
-      if (obj instanceof Item) {
-        Item that = (Item) obj;
-        return Objects.equals(this.ns, that.ns)
-            && Objects.equals(this.key, that.key)
-            && Objects.equals(this.type, that.type)
-            && Objects.equals(this.value, that.value)
-            && Objects.equals(this.shortValue, that.shortValue)
-            && Objects.equals(this.label, that.label)
-            && Objects.equals(this.url, that.url);
+    private static FormattedItemValue safeFormat(Type type, @Nullable Object value) {
```

Comment:
I think we could put this code in the `Type#format`. That would allow a type-specific default?